jujugui/static/gui/src/app/components/entity-header/entity-header.js

```diff
@@ -55,7 +55,24 @@ YUI.add('entity-header', function() {
       @param {Object} e The click event
     */
     _handleDeployClick: function(e) {
-      this.props.deployService(this.props.entityModel);
+      var entityModel = this.props.entityModel;
+      var entity = entityModel.toEntity();
+      if (entity.type === 'charm') {
+        this.props.deployService(entityModel);
+        this._closeEntityDetails();
+      } else {
+        var id = entity.id.replace('cs:', '');
```

Comment:
The entity id is something like 'cs:~charmers/django' but when we use that to do a lookup in the api we insert it into the url like this `https://api.jujucharms.com/charmstore/v4/~charmers/django...` hence needing to drop the `cs:` before generating the URL.